network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -97,15 +111,174 @@ public double getI() {
         return isConnected() ? Math.hypot(getP(), getQ()) / (Math.sqrt(3.) * getBusView().getBus().getV() / 1000) : 0;
     }
 
+    private Resource<VoltageLevelAttributes> getVoltageLevelResource() {
+        return index.getStoreClient().getVoltageLevel(index.getNetwork().getUuid(), attributes.getVoltageLevelId())
+                .orElseThrow(IllegalStateException::new);
+    }
+
+    private Set<Integer> getBusbarSectionNodes(Resource<VoltageLevelAttributes> voltageLevelResource) {
+        return index.getStoreClient().getVoltageLevelBusbarSections(index.getNetwork().getUuid(), voltageLevelResource.getId())
+                .stream().map(resource -> resource.getAttributes().getNode())
+                .collect(Collectors.toSet());
+    }
+
+    private static Graph<Integer, Edge> filterSwitches(Graph<Integer, Edge> graph, Predicate<SwitchAttributes> filter) {
+        return new AsSubgraph<>(graph, null, graph.edgeSet()
+                .stream()
+                .filter(edge -> {
+                    if (edge.getBiConnectable() instanceof SwitchAttributes) {
+                        SwitchAttributes switchAttributes = (SwitchAttributes) edge.getBiConnectable();
+                        return filter.test(switchAttributes);
+                    }
+                    return true;
+                })
+                .collect(Collectors.toSet()));
+    }
+
+    private boolean connectNodeBreaker(Resource<VoltageLevelAttributes> voltageLevelResource) {
+        boolean done = false;
+
+        Graph<Integer, Edge> graph = NodeBreakerTopology.INSTANCE.buildGraph(index, voltageLevelResource, true);
+        Set<Integer> busbarSectionNodes = getBusbarSectionNodes(voltageLevelResource);
+
+        // exclude open disconnectors to be able to calculate a shortest path with no open disconnector as we don't
+        // want to move disconnectors
+        Predicate<SwitchAttributes> closedDisconnector = switchAttributes -> !(switchAttributes.getKind() != SwitchKind.BREAKER && switchAttributes.isOpen());
+        Graph<Integer, Edge> filteredGraph = filterSwitches(graph, closedDisconnector);
```

Comment:
The predicat should not be (openable) : `BREAKER && notFictitious && isOpen`

Reply:
Here we want to exclude from the graph, everything that is open and not actionnable to avoid finding a path that contains such element so I think apart fictitious this is correct.